pkg/crdutil/crdutil.go

```diff
@@ -108,7 +120,17 @@ func walkCrdsDir(ctx context.Context, crdClient v1.CustomResourceDefinitionInter
 			return nil
 		})
 		if err != nil {
-			return fmt.Errorf("walk the path %s: %w", crdsDir, err)
+			return fmt.Errorf("walk the path %s: %w", crdsDirs, err)
+		}
+	}
+	return nil
+}
+
+func applyCrdFiles(ctx context.Context, crdClient v1.CustomResourceDefinitionInterface) error {
```

Comment:
For the `applyCRD` call, err can be intermediate (i.e other than NOT_FOUND when the CRD is already created) and curCRD can be nil. need to handle that case.

Reply:
Good catch 👍🏻